In [2]:
#Imnporting libraries
import pandas as pd
import numpy as np
import os

#Listing the existing files in the dir
for root, dirs, files in os.walk("."):  
    for filename in files:
        print(filename)

1st_assgnmt_collab.ipynb
1st_assignment_intro.ipynb
1st_assignment_matrix.ipynb
2nd_assignment_collab.ipynb
2nd_assignment_intro.ipynb
3rd_assignment_evaluation_metrics.ipynb
README.md
COMMIT_EDITMSG
config
description
FETCH_HEAD
HEAD
index
ORIG_HEAD
applypatch-msg.sample
commit-msg.sample
fsmonitor-watchman.sample
post-update.sample
pre-applypatch.sample
pre-commit.sample
pre-push.sample
pre-rebase.sample
pre-receive.sample
prepare-commit-msg.sample
update.sample
exclude
HEAD
master
master
a6f461b1cda82bc2bc64e2f8e752c594cb4b79
5889fd52bdb928eb7b8ef26232706c784a7fb1
8a81d4f821e4a1d51a8c68599d74959a23bb7d
ff03ce761e5a44ccc2b2a825bf574c1bee0631
44966c920b4d862ad58a805a0ece2382922184
05b0aaa2d7ebb03a3e0fb3d246cda98a5b004a
96b16af7d370ec7985b75b69d3af36a913954d
2974eb78edd82b431c3690f7464a503c694356
2041e1411e121aea1a74646ec651e7c9c954db
807046e3874e227078d513b2e90156c2ab1414
fdc207848247f45b640d1b9e61f889fd73a974
5a1c6f15253eed4ae0e7c623e7d5f649cdd8e4
9dd7992c843ab1040b900da7e3335ed0ff55

In [13]:
#Loading the datasets
users = pd.read_csv('./files/users.csv').set_index('User')
items = pd.read_csv('./files/items.csv').set_index('Unnamed: 0')
items_set = items.iloc[2:,:16]

In [23]:
frst_feature = items_set.iloc[:,1].sort_values(ascending=False).head(5); frst_feature
snd_feature = items_set.iloc[:,2].sort_values(ascending=False).head(5); snd_feature

Unnamed: 0
14     0.198715
680    0.189565
24     0.181570
275    0.161559
38     0.161058
Name: 2, dtype: float64

Task: For the first 2 features, provide the top 5 movies for that feature. Only provide the movie IDs.

Solution: For the first feature, the top movies are 4327 (0.281990), 414 (0.218089), 3049 (0.190879), 8467 (0.190638) and 854 (0.158601).
For the 2nd feature, the top five movies are 14 (0.198715), 680 (0.189565), 24 (0.181570), 275 (0.161559), and 38 (0.161058).

In [139]:
#Calculating top five movies for user 4469.
target = users.loc[4469,:]
target_df = pd.DataFrame(target)
weight_df = items.iloc[0,:].to_frame(name='Weight').iloc[1:,:].drop('Unnamed: 17').drop('Unnamed: 18')
merged = pd.DataFrame(target_df).join(weight_df).join(items_set.T)
pred = merged.iloc[:,2:].multiply(merged.iloc[:,0], axis=0).multiply(merged.iloc[:,1], axis=0).sum(axis=0).T.add(merged.iloc[:,0].sum()).sort_values(ascending=False).head(5); pred

278    0.201508
453    0.177114
98     0.167440
238    0.166008
13     0.164573
dtype: float64

Task: Compute the top 5 movies for user 4469. Only provide the movie IDs of the 5 recommendations.

Solution: Movies 278, 453, 98, 238 and 13 were top five movies for user 4469.